brainframe_qt/ui/resources/video_items/streams/stream_listener_widget.py

```diff
@@ -88,61 +53,72 @@ def handle_stream_url(stream_url: Optional[str]) -> None:
             if stream_url is None:
                 return
 
+            # User must have already changed stream again by the time this callback is
+            # called. Just forget about this current change request.
+            if stream_conf is not self.stream_conf:
+                return
+
             self._subscribe_to_stream(stream_conf, stream_url)
 
         QTAsyncWorker(self, self._get_stream_url, f_args=(stream_conf,),
                       on_success=handle_stream_url) \
             .start()
 
-    def _clear_current_stream_reader(self):
-        """If we currently have a stream reader, unsubscribe its listener
-        and clear any posted events"""
-
-        # Ensure that we're not storing a stream_conf
-        self.stream_conf = None
-
-        if not self.stream_reader:
-            return
-
-        self.destroyed.disconnect()
-
-        self.stream_reader.remove_listener(listener=self)
-
-        # Make sure no more events are sent to this listener
-        QCoreApplication.removePostedEvents(self)
+    def _disconnect_stream_reader(self) -> None:
```

Comment:
I noticed in the diff we had `QCoreApplication.removePostedEvents(self)`. Is something similar necessary?

Reply:
No longer necessary. Qt handles that logic for us automatically